cvat/apps/iam/permissions.py

```diff
@@ -365,14 +365,20 @@ class LambdaPermission(OpenPolicyAgentPermission):
     def create(cls, request, view, obj):
         permissions = []
         if view.basename == 'function' or view.basename == 'request':
-            for scope in cls.get_scopes(request, view, obj):
+            scopes = cls.get_scopes(request, view, obj)
+            for scope in scopes:
                 self = cls.create_base_perm(request, view, scope, obj)
                 permissions.append(self)
 
-            task_id = request.data.get('task')
-            if task_id:
-                perm = TaskPermission.create_scope_view_data(request, task_id)
+            job_id = request.data.get('job')
+            if job_id:
+                perm = JobPermission.create_scope_view_data(request, job_id)
                 permissions.append(perm)
+            else:
+                task_id = request.data.get('task')
+                if task_id:
+                    perm = TaskPermission.create_scope_view_data(request, task_id)
+                    permissions.append(perm)
```

Comment:
Ok. I need to note, that it can be done in a lot of places in the codebase, but we can definitely do this in the new code since we are targeting 3.8 and above on the server. SDK and CLI will have to wait for 3.7 EOL.

Reply:
Sure; I'm not suggesting it for SDK/CLI.